### PR TITLE
docs: improve the identity data model page

### PR DESCRIPTION
**Pull request by zepatrik**
As some parts of that page were not clear I tried to improve it. Please correct me wherever I missed some content.

### CODE REVIEW

File: docs/docs/concepts/identity-data-model.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-id: identity-user-model
+id: identity-data-model
```
[the remark this answers is not in the document]
Comment (zepatrik) [reply docs/docs/concepts/identity-data-model.md:2]: ah ok I'll revert it then

[the remark this answers is not in the document]
Comment (zepatrik) [reply docs/docs/concepts/identity-data-model.md:2]: thought it's another instance of "user is identity says identity's identity"

[the remark this answers is not in the document]
Comment (aeneasr) [reply docs/docs/concepts/identity-data-model.md:2]: SEO :)

File: docs/docs/concepts/index.md [hunk 2]
```diff
@@ -262,7 +262,7 @@ system:
   attached to the profile.
 
 ORY Kratos implements both scenarios by using
-[JSON Schemas for Identity Traits](./identity-user-model.md)
+[JSON Schemas for Identity Traits](./identity-data-model.md)
```
Comment (aeneasr) [inline docs/docs/concepts/index.md:265]:
```suggestion
[JSON Schemas for Identity Traits](./identity-user-model.md)
```

File: docs/sidebar.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ module.exports = {
   Concepts: [
     "concepts/index",
     "concepts/ui-user-interface",
-    "concepts/identity-user-model",
+    "concepts/identity-data-model",
```
Comment (aeneasr) [inline docs/sidebar.js:8]: revert

File: docs/docs/self-service/strategies/username-email-password.mdx [hunk 4]
```diff
@@ -45,7 +45,7 @@ identity:
 ```
 
 If you don't know what that means, please read the
-["Concepts: Identity" Chapter](../../concepts/identity-user-model.md).
+["Concepts: Identity" Chapter](../../concepts/identity-data-model.md).
```
Comment (aeneasr) [inline docs/docs/self-service/strategies/username-email-password.mdx:48]: revert

File: docs/docs/self-service/strategies/user-settings-profile.md [hunk 5]
```diff
@@ -7,7 +7,7 @@ The `profile` strategy allows a user to change their identity traits
 ("profile").
 
 The updated traits must be valid against the JSON Schema defined for its
-[Identity Traits](../../concepts/identity-user-model.md). If one or more fields
+[Identity Traits](../../concepts/identity-data-model.md). If one or more fields
```
Comment (aeneasr) [inline docs/docs/self-service/strategies/user-settings-profile.md:10]: revert

File: docs/docs/sdk/api.md [hunk 6]
```diff
@@ -1291,7 +1291,7 @@ Accept: application/json
 This endpoint deletes an identity. This can not be undone.
 
 Learn how identities work in
-[ORY Kratos' User And Identity Model Documentation](https://www.ory.sh/docs/next/kratos/concepts/identity-user-model).
+[ORY Kratos' User And Identity Model Documentation](https://www.ory.sh/docs/next/kratos/concepts/identity-data-model).
```
Comment (aeneasr) [inline docs/docs/sdk/api.md:1294]: revert

File: docs/docs/sdk/api.md [hunk 7]
```diff
@@ -1051,7 +1051,7 @@ The full identity payload (except credentials) is expected. This endpoint does
 not support patching.
 
 Learn how identities work in
-[ORY Kratos' User And Identity Model Documentation](https://www.ory.sh/docs/next/kratos/concepts/identity-user-model).
+[ORY Kratos' User And Identity Model Documentation](https://www.ory.sh/docs/next/kratos/concepts/identity-data-model).
```
Comment (aeneasr) [inline docs/docs/sdk/api.md:1054]: revert

**Pull request merged**